containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceListener.java

```diff
@@ -0,0 +1,158 @@
+                    URL url = new URL(NamespaceHandler.KUBERNETES_MASTER_URL + "/" + path + "?watch=true");
+
+                    HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+                    connection.setRequestMethod("GET");
+
+                    connection.setRequestProperty("Accept", "application/json");
+                    connection.setRequestProperty("Authorization", "Bearer " + token);
+
+                    connection.setConnectTimeout(5000);
+
+                    // Since the connection is left open for new events, set to 0 which indicates an infinite timeout
+                    // Note that the server will eventually close this connection, so we will need to reconnect when this happens
+                    connection.setReadTimeout(0);
+
+                    connection.connect();
+                    int responseCode = connection.getResponseCode();
+
+                    if (responseCode == 200) {
+
+                        JsonFactory jsonFactory = new JsonFactory();
+                        jsonFactory.setCodec(new ObjectMapper());
+
+                        InputStream inputStream = connection.getInputStream();
+                        JsonParser jsonParser = jsonFactory.createParser(inputStream);
+
+                        // this will fetch all the available tenants and will then wait until new tenant events occur
+                        JsonNode jsonNode = jsonParser.readValueAsTree();
+
+                        // When the connection is closed, the response is null
+                        while (jsonNode != null) {
+                            JsonNode metaData = jsonNode.get("object").get("metadata");
+                            String projectName = metaData.get("name").asText();
+                            String projectID = metaData.get("uid").asText();
+
+                            String action = jsonNode.get("type").asText();
+
+                            if (action.equals("ADDED")) {
+                                log.debug("Project " + projectName + "[" + projectID + "] has been added.");
+                                namespaces.put(projectName, projectID);
+                            } else if (action.equals("DELETED")) {
+                                log.debug("Project " + projectName + "[" + projectID + "] has been removed.");
+                                namespaces.remove(projectName, projectID);
+                            }
+
+                            // this will read the next available tenant if available or wait until the next available
+                            // tenant event occurs
+                            jsonNode = jsonParser.readValueAsTree();
+                        }
+
+                        jsonParser.close();
+                        inputStream.close();
+                        namespaces.clear();
+
+                    } else {
+                        log.error("Error getting metadata. Response code " + responseCode);
```

Comment:
Use ```errorf``` here instead and get rid of the string concatenation. Actually I would use ```warnf```.